### PR TITLE
[Hub Generated] Review request for Microsoft.KeyVault to add version stable/2022-07-01

### PR DESCRIPTION
**Pull request by adarce**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/adarce/azure-rest-api-specs/dev-keyvault-Microsoft.KeyVault-2022-07-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify

  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
**By the end of the month (June 2022). Likely in a week or so.**

  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
**ETA: 2022-07-01**

  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by adarce** [comment]
@heaths and @weidongxu-microsoft : This is intended to be a simple PR that adds the changes from the existing preview control plane API version 2021-11-01-preview to the stable API version 2022-07-01. I used OpenAPI Hub to generate this PR, so I don't expect this PR to have introduced any new CI validation issues. How should the existing CI validation issues in this PR be handled?

**Comment by weidongxu-microsoft** [comment]
@adarce 

Is https://github.com/Azure/azure-rest-api-specs/pull/19395/commits/7f877afc6093b37e5a8ba70a8067e3d568381736 the only change made on 2021-11-01-preview (and it is just merged from GA 2021-10-01)?

---

`LRO_RESPONSE_HEADER` is a relative new check. To fix it, you need to add the LRO header in both swagger 202 response and that example.
Alternatively, you can suppress it in readme.

**Comment by adarce** [comment]
> Is https://github.com/Azure/azure-rest-api-specs/commit/7f877afc6093b37e5a8ba70a8067e3d568381736 the only change made on 2021-11-01-preview (and it is just merged from GA 2021-10-01)?

Yes, that's correct.

---

> LRO_RESPONSE_HEADER is a relative new check. To fix it, you need to add the LRO header in both swagger 202 response and that example.

Thanks, I've pushed a new commit to my PR which addresses this check. I don't think that it's a breaking change (the service always returned LRO headers - it just wasn't annotated correctly in the Swagger spec).

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json [hunk 1]
```diff
@@ -0,0 +1,1473 @@
...
+            "$ref": "./examples/ManagedHsm_Delete.json"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "ManagedHsms"
+        ],
+        "operationId": "ManagedHsms_Get",
+        "x-ms-long-running-operation": true,
```
Comment (raosuhas) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:229]: > "x-ms-long-running-operation": true, [](http://example.com/codeflow?start=7&length=37)

GET cannot be long running.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json [hunk 2]
```diff
@@ -0,0 +1,1473 @@
...
+            "description": "Retrieved managed HSM Pool",
+            "schema": {
+              "$ref": "#/definitions/ManagedHsm"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously.",
+            "headers": {
+              "Location": {
+                "description": "The URI to poll for completion status.",
```
Comment (raosuhas) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:260]: there should not be a 202 for GET

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-3](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-3)

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/common.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          "type": "string",
+          "description": "Error code. This is a mnemonic that can be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "User friendly error message. The message is typically localized and may vary with service version."
+        }
+      },
+      "x-ms-external": true,
+      "type": "object"
```
Comment (raosuhas) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/common.json:33]: I am curious , why did you need to add the type:object filed everywhere ?

**Comment by raosuhas** [comment]
Added comments. Please remove the "ARMChangesRequested" label when you want the API reviewer to look at this review again

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/common.json (hunk 3 above)
Comment (adarce) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/common.json:33]: They were added for parity with the previous stable ApiVersion (2021-10-01), which contains these type: object fields everywhere.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json (hunk 1 above)
Comment (adarce) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:229]: After confirming with the Managed HSM team, I've removed this from the GET operation.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json (hunk 2 above)
Comment (adarce) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:260]: After confirming with the Managed HSM team, I've removed this from the GET operation.

**Review by raosuhas — APPROVED** [review]
:shipit:

**Comment by riteshchitlangia1** [comment]
MHSM changes look good to me.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json [hunk 4]
```diff
@@ -0,0 +1,1597 @@
...
+        "name": {
+          "type": "string",
+          "description": "The managed hsm name."
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The parameters used to check the availability of the managed hsm name."
+    },
```
Comment (weidongxu-microsoft) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:1490]: keyvault defined the payload slightly different.
https://github.com/Azure/azure-rest-api-specs/blob/ff63535af8d98b47d1a57aa9796cba4ae070e393/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/keyvault.json#L1685-L1709

I don't think there is requirement from ARM on the schema, so it is up to you.

Comment (adarce) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:1490]: @neeerajaakula or @riteshchitlangia1 : Please confirm that the slight discrepancies here are acceptable for Managed HSM.

**Comment by adarce** [comment]
@weidongxu-microsoft : AKV's service has been deployed to all Public cloud regions, and our ARM manifest has also been applied everywhere. Can you please merge this PR?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json (hunk 4 above)
Comment (neeerajaakula) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2022-07-01/managedHsm.json:1490]: This is not relevant for Managed HSM api
